ggml-metal.m

```diff
@@ -224,56 +224,47 @@ bool ggml_metal_add_buffer(
         }
 
         size_t page_size = getpagesize();
-        size_t aligned_size = size;
-        if ((aligned_size % page_size) != 0) {
-            aligned_size += (page_size - (aligned_size % page_size));
-        }
-
-        ctx->buffers[ctx->n_buffers].name = name;
-        ctx->buffers[ctx->n_buffers].data = data;
-        ctx->buffers[ctx->n_buffers].size = size;
+        size_t sys_max_buffer_size = 2ul * 1024ul * 1024ul * 1024ul; // ctx->device.maxBufferLength;
```

Comment:
Note: this is an artificial 2GB limit that I had in place to test this out, given I don't actually bump into maxBufferLength on my M1 Max. 

Should be switched back to ctx->device.maxBufferLength once issues are worked out.